lang/cs/Org.Apache.REEF.Tang/Formats/AvroConfigurationSerializer.cs

```diff
@@ -260,6 +260,28 @@ public AvroConfiguration ToAvroConfiguration(IConfiguration c)
                 l.Add(new ConfigurationEntry(e.Key.GetFullName(), val));
             }
 
+            IEnumerator bl = conf.GetBoundList().GetEnumerator();
+            while (bl.MoveNext())
```

Comment:
why not a for each loop?

Reply:
What is the expected behavior for an empty list? The code does run - however nothing is written out because there is nothing to serialize. If that is expected, I will modify the test accordingly.

---
In reply to: [223779318](https://github.com/apache/reef/pull/1479#discussion_r223779318) [](ancestors = 223779318)

Reply:
Yes, I believe that is the expected behavior.